### ISSUE TITLE
2021-11-2

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#643]
241左边是一样的，右边多了一个momentum encoder, 然后生成一个队列; 据说是模板? 
242代码里的 k 可以看作模板，q 看作查询元素，每一个输入未知图像的特征由 f_q 提取， 现在给一系列由 f_k 提取的模板特征 (比如狗的特征、猫的特征) ，就能使用 f_q 与 f_k 的度量值来确定 f_q 是属于什么。
243(1) 现在我们有一堆无标签的数据，拿出一个 Batch，代码表示为 x，也就是 [公式] 张图片，分别进行两种不同的数据增强，得到 x_q 和 x_k，则 x_q 是 [公式] 张图片，x_k 也是 [公式] 张图片。
244(2) x_q 通过 Encoder 得到特征 q，维度是 [公式] ，这里特征空间由一个长度为 [公式] 的向量表示。// x_q 通过 Momentum Encoder 得到特征 k，维度是 [公式]
245(3) Momentum Encoder的参数不更新：# 不使用梯度更新f_k的参数，这是因为文章假设用于提取模板的表示应该是稳定的，不应立即更新
246(4) 计算 [公式] 张图片的自己与自己的增强图的特征的匹配度; (2021-10-21)
247(5) 计算 [公式] 张图片与队列中的 [公式] 张图的特征的匹配度; (2021-10-21)
248(6) 把 4, 5 两步得到的结果concat起来：; 这里得到的 logits 的维度是 (N, K+1)，把它看成是一个矩阵的话呢，有 N 行，代表一个 Batch里面的 N 张图片。每一行的第1个元素是某张图片自己与自己的匹配度，每一行的后面K 个元素是某张图片与其他 K 个图片的匹配度，如下图9所示，图9展示的是某一行的信息，这里的 K=2。(2021-10-21)
249(7) NCE损失函数，就是为了保证自己与自己衍生的匹配度输出越大越好，否则越小越好;
